src/features/broker-protection.js

```diff
@@ -0,0 +1,21 @@
+import ContentFeature from '../content-feature.js'
+import { execute } from './broker-protection/execute.js'
+
+export default class BrokerProtection extends ContentFeature {
+    init () {
+        this.messaging.subscribe('onActionReceived', (/** @type {any} */params) => {
```

Comment:
We shouldn't be worried about a possible race condition here, right?
Say, if Native decides to requests an action, but the page navigates, and the action is delivered to the new page that Native did not intend to ask.
I'm not familiar with the DBP use-case, but this was a big issue for autoconsent last year

Reply:
It's a great question, something I raised at the beginning of the project. 

The way the scheduler is implemented on the native side means we were able to shrink it down to just this implementation.